File: aat-android/src/main/java/ch/bailu/aat/util/sql/AndroidDbResultSet.java
package ch.bailu.aat.util.sql;

import android.database.Cursor;

import ch.bailu.aat_lib.util.sql.DbResultSet;

public class AndroidDbResultSet implements DbResultSet {
    private final Cursor cursor;

    public AndroidDbResultSet(Cursor cursor) {
        this.cursor = cursor;
    }

    @Override
    public boolean moveToFirst() {
        return cursor.moveToFirst();
    }

    @Override
    public boolean moveToNext() {
        return cursor.moveToNext();
    }

    @Override
    public void close() {
        cursor.close();
    }


    @Override
    public boolean moveToPrevious() {
        return cursor.moveToPrevious();
    }

    @Override
    public boolean moveToPosition(int pos) {
        return cursor.moveToPosition(pos);
    }

    @Override
    public int getCount() {
        return cursor.getCount();
    }

    @Override
    public String getString(String column) {
        return cursor.getString(cursor.getColumnIndex(column));
    }

    @Override
    public long getLong(String column) {
        return cursor.getLong(cursor.getColumnIndex(column));
    }

    @Override
    public float getFloat(String column) {
        return cursor.getFloat(cursor.getColumnIndex(column));
    }

    @Override
    public int getPosition() {
        return cursor.getPosition();
    }

    @Override
    public boolean isClosed() {
        return cursor.isClosed();
    }
}

File: aat-android/src/main/java/ch/bailu/aat/util/sql/AndroidDbResultSet.kt
package ch.bailu.aat.util.sql

import android.database.Cursor
import ch.bailu.aat_lib.util.sql.DbResultSet

class AndroidDbResultSet(private val cursor: Cursor) : DbResultSet {
    override fun moveToFirst(): Boolean {
        return cursor.moveToFirst()
    }

    override fun moveToNext(): Boolean {
        return cursor.moveToNext()
    }

    override fun close() {
        cursor.close()
    }

    override fun moveToPrevious(): Boolean {
        return cursor.moveToPrevious()
    }

    override fun moveToPosition(pos: Int): Boolean {
        return cursor.moveToPosition(pos)
    }

    override fun getCount(): Int {
        return cursor.count
    }

    override fun getString(column: String): String {
        val index = cursor.getColumnIndex(column)
        return if (index > -1) cursor.getString(index) else ""
    }

    override fun getLong(column: String): Long {
        val index = cursor.getColumnIndex(column)
        return if (index > -1) cursor.getLong(index) else 0
    }

    override fun getFloat(column: String): Float {
        val index = cursor.getColumnIndex(column)
        return if (index > -1) cursor.getFloat(index) else 0f
    }

    override fun getPosition(): Int {
        return cursor.position
    }

    override fun isClosed(): Boolean {
        return cursor.isClosed
    }
}
